### PR TITLE
docs: update commit message guideline

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
Updated commit message guideline in CONTRIBUTING.md
Moved CONTRIBUTING.md to `.github` directory

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [ ] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [ ] I tested non-trivial changes on a real device
- [ ] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Why move it inside `.github`? If moving it is necessary, should also update link in `README.md`

**Comment by imaginaris** [comment]
> Why move it inside `.github`? If moving it is necessary, should also update link in `README.md`

According to this: https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors
The `CONTRIBUTING.md` file should sit in docs or .github dirs BUT it seems like it works in root directory as well. I'll move it back where it was and update iOS repo instead

**Comment by vincentdelmundo** [comment]
> > Why move it inside `.github`? If moving it is necessary, should also update link in `README.md`
> 
> According to this: https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors The `CONTRIBUTING.md` file should sit in docs or .github dirs BUT it seems like it works in root directory as well. I'll move it back where it was and update iOS repo instead

@imaginaris
Got it.
Either way is fine with me, just want to know the reason :)

**Review by vincentdelmundo — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -17,17 +17,17 @@ Please include unit tests where necessary to cover any functionality that is int
 * All Java code must follow [Google's Java Code style](https://google.github.io/styleguide/javaguide.html), the only exception being that annotations on members or classes may be on the same line (no forced line break).
 
 ## Commit messages
-Each commit message consists of a header, a body and a footer. The header has a special format that includes a type and a subject:
+Each commit message consists of a header, and an optional body and footer. The header has a special format that includes a type, a subject and a optional ticket number:
```
Comment (leila-gu) [inline CONTRIBUTING.md:20]: typo:   a optional ticket number -> an optional ticket number

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -40,29 +40,58 @@ If the commit reverts a previous commit, it should begin with `revert: `, follow
...
+* _**Pull request #519:**_ improve: refactor user data handling (SDKCF-3970)
+<br>Remove default prefixes added by Bitbucket.
+ 
+* improve: refactor user data handling (SDKCF-3970) _**(#23)**_
+<br>Remove default suffixes added by Github.
+ 
+* _**Revert "refactor user data handling"**_
+<br>Revert should start with `revert:` followed by exact original commit message of reverted commit (without quotes).
+ 
+* fix: _**fix crash on iOS 15**_ (SDKCF-0000)
```
Comment (vincentdelmundo) [inline CONTRIBUTING.md:78]: maybe make this as "Android 12"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -40,29 +40,58 @@ If the commit reverts a previous commit, it should begin with `revert: `, follow
...
+<br>Commit message should contain some context. In this case - a cause of the crash.
+ 
+* feat: _**add new build configuration for users that require user data to be stored in another directory and add migration logic with exception of users migration from SDK version 1.0.0**_
+<br>Try to fit within 100 characters limit. Use abbreviations if necessary. Focus on the most important change.
+
+#### Good examples:
+* improve: refactor user data handling (SDKCF-3970)
+* improve: refine URL session configuration, add timeouts (SDKCF-0000)
+* docs: update README with SPM information (SDKCF-4388)
```
Comment (vincentdelmundo) [inline CONTRIBUTING.md:87]: we can change this from "SPM" to "gradle"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -40,29 +40,58 @@ If the commit reverts a previous commit, it should begin with `revert: `, follow
...
+* chore: bump RSDKUtils to 2.1.0 (SDKCF-4392)
+* chore: prepare 5.0.0 release (SDKCF-4160)
+* revert: improve: refine URL session configuration, add timeouts (SDKCF-0000)
+<br>This reverts commit 6608beb114610ec0a5e96b77e62a5590753e247e.
+* feat: add support for id tracking (SDKCF-4072)
+* feat: [iOS][Analytics] add "Delete user data" button (SDKCF-4072)
```
Comment (vincentdelmundo) [inline CONTRIBUTING.md:93]: we can remove this

**Review by vincentdelmundo — APPROVED** [review]
LGTM. Just minor comments.
Please check.

**Review by leila-gu — COMMENTED** [review]
LGTM, only one small comment.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -40,29 +40,58 @@ If the commit reverts a previous commit, it should begin with `revert: `, follow
...
+* feat: _**add new build configuration for users that require user data to be stored in another directory and add migration logic with exception of users migration from SDK version 1.0.0**_
+<br>Try to fit within 100 characters limit. Use abbreviations if necessary. Focus on the most important change.
+
+#### Good examples:
+* improve: refactor user data handling (SDKCF-3970)
+* improve: refine URL session configuration, add timeouts (SDKCF-0000)
+* docs: update README with SPM information (SDKCF-4388)
+* chore: bump RSDKUtils to 2.1.0 (SDKCF-4392)
+* chore: prepare 5.0.0 release (SDKCF-4160)
+* revert: improve: refine URL session configuration, add timeouts (SDKCF-0000)
```
Comment (MdUsmanGani) [inline CONTRIBUTING.md:90]: Can we use two <**type**> together?

**Review by MdUsmanGani — APPROVED** [review]
LGTM

File: CONTRIBUTING.md (hunk 5 above)
Comment (imaginaris) [reply CONTRIBUTING.md:90]: Yes. That's an exception for revert commits. It starts with `revert:` followed by exact message of reverted commit. Maybe we could put the reverted commit message in "", I'm not sure

**Pull request merged**